### ISSUE TITLE
Query parameters are not percent-encoded for curl in OAS3, but are in Swagger 2

### ISSUE THREAD

**Issue by cklos1007** [swagger-api/swagger-ui#5304]
### Q&A (please complete the following information)
 - OS: [macOS]
 - Browser: [chrome]
 - Version: [73.0.3683.86]
 - Method of installation: [npm]
 - Swagger-UI version: [3.22.0]
 - Swagger/OpenAPI version: [Swagger 2.0, OpenAPI 3.0]

### Content & configuration
Swagger 2.0 YAML
```yaml
swagger: '2.0'
info:
  title: testing Swagger 2.0
  version: '3.0'
paths:
  /info:
    get:
      parameters:
       - in: query
         name: page[offset]
         type: integer
       - in: query
         name: page[limit]
         type: integer
      responses:
        '200':
          description: OK
```

OpenAPI 3.0 YAML
```yaml
openapi: 3.0.0
info:
  title: testing OAS 3.0
  version: '3.0'
paths:
  /info:
    get:
      parameters:
       - in: query
         name: page[offset]
         schema:
           type: integer
       - in: query
         name: page[limit]
         schema:
            type: integer
      responses:
        '200':
          description: OK
```

### Describe the bug you're encountering
When defining query parameters that contain reserved characters, Swagger UI encodes the reserved characters in Swagger 2.0, but does not do so for OpenAPI 3.0

### To reproduce...

Steps to reproduce the behavior for each YAML:
1. Click on 'Try It Out'
2. Enter an integer for 'page[offset]'
3. Enter an integer for 'page[limit]'
4. Click on 'Execute'

### Screenshots
Look at the curl command for Swagger 2.0, which encodes the page[offset] and page[limit] correctly:
![swagger2](https://user-images.githubusercontent.com/49659019/56165304-33309380-5f98-11e9-8ed0-56e050305d22.png)

Look at the curl command for OAS 3.0, which does not:
![OAS3](https://user-images.githubusercontent.com/49659019/56165282-2a3fc200-5f98-11e9-82bb-01cd829ca32b.png)

### Additional context or thoughts
Curl requires that the url has all reserved characters percent-encoded.  When I try out APIs with these type of query parameters in OAS 3.0 using the Swagger UI "Try It Out", I get failures because the square brackets in these query parameter names are not properly encoded.

Either fix this bug or allow OAS 3.0 Swagger UI to configure a setting to enforce encoding when forming the URL for the curl command.

**Comment by sm2017** [comment]
I have same issue

**Comment by john3300** [comment]
I have fixed this issue with swagger-api/swagger-js#1487 which is available in release v3.25.0.

**Comment by cklos1007** [comment]
Closing this issue as it was fixed with swagger-api/swagger-js#1487 which is available in release v3.25.0.
